basiclibrary/src/main/java/basiclibrary/Library.java

```diff
@@ -189,4 +192,43 @@ public String tally(List<String> input){
         String output = winner + " received the most votes!";
         return output;
     }
+
+    //linter
+    public String jsLinter(String path) {
+        int lineNumber = 0;
+        int errorNumber = 0;
+        String outputMessage = "";
+        File fileToRead = new File(path);
+        ArrayList<Integer> errorList = new ArrayList<>();
+
+        try{
+            Scanner jsScanner = new Scanner(fileToRead);
+
+            while(jsScanner.hasNextLine()){
+                ++lineNumber;
+                String checkLine = jsScanner.nextLine();
+
+                if( ( !checkLine.startsWith("//") && !checkLine.endsWith("{") && !checkLine.endsWith("}") && !checkLine.isEmpty() & !checkLine.contains("if") && !checkLine.contains("else") ) && !checkLine.endsWith(";") ){
```

Comment:
If you have a really long conditional statement you can always put each conditional on a separate line to make it more readable (100% your choice though).

`if ( !checkLine.startsWith("//") 
    && !checkLine.endsWith("{") 
    && !checkLine.endsWith("}") 
    && !checkLine.isEmpty() 
    && !checkLine.contains("if") 
    && !checkLine.contains("else") 
    && !checkLine.endsWith(";") ) {`

Also, it looks like you're missing a `&` after `.isEmpty()`

Reply:
Each conditional above should be indented, GitHub won't let me indent though :-(